system/Debug/Exceptions.php

```diff
@@ -158,17 +158,22 @@ public function exceptionHandler(Throwable $exception)
      */
     public function errorHandler(int $severity, string $message, ?string $file = null, ?int $line = null)
     {
-        if ($this->isDeprecationError($severity) && $this->config->logDeprecationsOnly) {
-            return $this->handleDeprecationError($message, $file, $line);
-        }
-
-        if (error_reporting() & $severity) {
+        if ($this->isDeprecationError($severity)) {
             // @TODO Remove if Faker is fixed.
-            if ($this->isFakerDeprecationError($severity, $message, $file, $line)) {
-                // Ignore the error.
+            if ($this->isFakerDeprecationError($message, $file, $line)) {
                 return true;
             }
 
+            if (! $this->config->logDeprecationsOnly
+                || filter_var(env('CODEIGNITER_SCREAM_DEPRECATIONS'), FILTER_VALIDATE_BOOLEAN)
```

Comment:
This seems unduly specific and conflicts with your statement of using any truthy value. How about simply:
```suggestion
                || env('CODEIGNITER_SCREAM_DEPRECATIONS')
```